h2o-algos/src/main/java/hex/tree/gbm/GBM.java

```diff
@@ -184,6 +184,12 @@
     }
   }
 
+  @Override
+  protected void validateMaxDepth() {
+    if (_parms._max_depth < 0) error ("_max_depth", "_max_depth must be >= 0.");
+    if(_parms._max_depth == 0) _parms._max_depth = Integer.MAX_VALUE;
```

Comment:
I would set to unlimited as soon as `max_depth <= 0`, raising on negative looks unnecessary.